agent/connect/ca/provider_vault_test.go

```diff
@@ -212,6 +212,52 @@ func TestVaultCAProvider_RenewToken(t *testing.T) {
 	})
 }
 
+func TestVaultCAProvider_RenewTokenStopWatcherOnConfigure(t *testing.T) {
+
+	SkipIfVaultNotPresent(t)
+
+	testVault, err := runTestVault(t)
+	require.NoError(t, err)
+	testVault.WaitUntilReady(t)
+
+	// Create a token with a short TTL to be renewed by the provider.
+	ttl := 1 * time.Second
+	tcr := &vaultapi.TokenCreateRequest{
+		TTL: ttl.String(),
+	}
+	secret, err := testVault.client.Auth().Token().Create(tcr)
+	require.NoError(t, err)
+	providerToken := secret.Auth.ClientToken
+
+	provider, err := createVaultProvider(t, true, testVault.Addr, providerToken, nil)
+	require.NoError(t, err)
+
+	var gotStopped = false
+	provider.stopWatcher = func() {
+		gotStopped = true
+	}
```

Comment:
Is this callback will called from another goroutine? I think it is, which probably means this would fail the race detector (even though bool assign is atomic on some CPUs anyway).

Could replace it with:
```suggestion
	var gotStopped = uint32(0)
	provider.stopWatcher = func() {
		atomic.StoreUint32(&gotStopped, 1)
	}
```

then later `require.Equal(t, uint32(1),  atomic.LoadUint32(&gotStopped))`

Reply:
It shouldn't be called as part of a goroutine (at least in this test) but is a call back from `provider.Configure.` Using atomic should make it more robust if that changes in the future.